packages/next/build/index.ts

```diff
@@ -806,13 +806,36 @@ export default async function build(dir: string, conf = null): Promise<void> {
       if (isSsg) {
         // For a non-dynamic SSG page, we must copy its data file from export.
         if (!isDynamic) {
-          await moveExportedPage(page, file, true, 'json')
+          if (additionalSsgPaths.has(page)) {
+            // TODO: maybe we should move it to a separate function? see duplicated code below
```

Comment:
+1 agreed! What function signature would you propose?

Reply:
`async addExtraRoutesForPage(page:string, extraRoutes:string[]): Promise<void>`

It will be function with side effects, as it will change finalPrerenderRoutes internally